useIsShake.js

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from 'react';
+import { Accelerometer } from 'expo-sensors';
+import { QUOTES_IN_DURATION, QUOTES_IN_DELAY, QUOTES_OUT_DURATION, SVG_LOAD_DURATION, SVG_OUT_DURATION, SVG_IN_DURATION, SVG_IN_DELAY, TRANSITION_TIME_2 } from './constants';
+
+export function useIsShake() {
+  const [{ x }, setData] = useState({
+    x: 0,
+  });
+  const [subscription, setSubscription] = useState(null);
+  const [toggle, setToggle] = useState(false);
+
+  let polls = [null, null];
+  let diffs = [];
+  let shakeCount = 0;
+  let isToggleReady;
+  let hasToggledBefore;
+
+  function isShake({ x: newX }) {
+    if (isToggleReady === true) {
+      if (diffs.length === 2) {
+        shakeCount++;
+        setToggle(true);
+        isToggleReady = false;
+
+        if (hasToggledBefore === undefined) hasToggledBefore = false;
+        else if (hasToggledBefore === false) hasToggledBefore = true;
```

Comment:
I don't feel great about how `hasToggledBefore` is handling three states: `undefined | false | true`. `hasToggledBefore` seems like it should either be `true` or `false`, and having the `undefined` here as an additional state is confusing. Let's think of a better way to describe these in the next PR